traits/trait_types.py

```diff
@@ -3336,6 +3338,145 @@ def as_ctrait(self):
         """
         return self.trait_maker.as_ctrait()
 
+# -------------------------------------------------------------------------------
+#  'None' trait:
+# -------------------------------------------------------------------------------
+
+
+class NoneTrait(TraitType):
+    """ Defines a trait that only accepts the None value
+    """
+
+    info_text = "None"
+
+    default_value = None
+
+    default_value_type = DefaultValue.constant
+
+    def __init__(self, **metadata):
+        default_value = metadata.pop("default", None)
+        if default_value is not None:
+            raise ValueError("Cannot set default value {} "
+                             "for NoneTrait".format(default_value))
+        super(NoneTrait, self).__init__(**metadata)
+
+    def validate(self, obj, name, value):
+        if value is None:
+            return value
+
+        self.error(obj, name, value)
+
+# -------------------------------------------------------------------------------
+#  'Union' trait:
+# -------------------------------------------------------------------------------
+
+
+class Union(TraitType):
+    """ Defines a trait whose value can be any of of a specified list of
+    trait types or list of trait type instances or None
+    """
+    # CTrait type map for special trait types:
+    type_map = {"event": TraitKind.event, "constant": TraitKind.constant}
+
+    def __init__(self, *traits, **metadata):
+        super(Union, self).__init__(traits, **metadata)
+        self.list_ctrait_instances = []
+        for trait in traits:
```

Comment:
We should check that there is at least one trait passed in and error if not present.

Reply:
`Either` allows having no attributes, It defaults to None, so I was letting `Union` do the same.

Reply:
When `Either` does it, it effectively is the same as `Trait(None)` which I think is different behaviour than what you are getting here.

I think with nothing passed in you will get a trait that can't be set to at all.  That's potentially OK as a design, but then you'll need special-casing in a number of places to make sure that things like the trait info make sense when you have no arguments, plus tests of those branches.

Reply:
Yes, that makes sense. I handled it by adding a `NoneTrait` as the argument if no arguments are passed in.